src/utils/web3React.ts

```diff
@@ -31,3 +31,22 @@ export const getLibrary = (provider): ethers.providers.Web3Provider => {
   library.pollingInterval = POLLING_INTERVAL
   return library
 }
+
+interface SignMessage {
+  provider: ethers.providers.JsonRpcProvider | ethers.providers.Web3Provider
+  message: string
+  account: string
+}
+
+/**
+ * BSC Wallet requires a different sign method
+ * @see https://docs.binance.org/smart-chain/wallet/wallet_api.html#binancechainbnbsignaddress-string-message-string-promisepublickey-string-signature-string
+ */
+export const signMessage = async ({ provider, account, message }: SignMessage): Promise<string> => {
```

Comment:
any reason why you pass an object instead of a list of params ?

Reply:
Yeah, I see what you mean. I originally did this because my first iteration had account optional. Since they are all required I think 3 separate arguments are better. Let me update.